core/src/context/mod.rs

```diff
@@ -88,6 +88,8 @@ pub struct Context {
     click_pos: (f32, f32),
 
     pub ignore_default_theme: bool,
+
+    removed_entities: HashSet<Entity>,
```

Comment:
The `EntityManager` in context actually already keeps track of which entities are 'alive'. You should be able to query it with the entity using the `is_alive` method.

Reply:
But entity IDs are recycled, aren't they? How can we tell the difference between an entity that was never destroyed and an entity that was destroyed and a new one built in its place?

Reply:
Entities have a generation part to them which is tracked by the id manager. So the entity id might be the same but the generation will be different.